flink/src/test/java/org/apache/iceberg/flink/data/TestFlinkOrcReaderWriter.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink.data;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.iceberg.Files;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.data.DataTest;
+import org.apache.iceberg.data.RandomGenericData;
+import org.apache.iceberg.data.Record;
+import org.apache.iceberg.data.orc.GenericOrcWriter;
+import org.apache.iceberg.flink.FlinkSchemaUtil;
+import org.apache.iceberg.io.CloseableIterable;
+import org.apache.iceberg.io.FileAppender;
+import org.apache.iceberg.orc.ORC;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.rules.TemporaryFolder;
+
+public class TestFlinkOrcReaderWriter extends DataTest {
+  private static final int NUM_RECORDS = 200;
+
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+
+  @Override
+  protected void writeAndValidate(Schema schema) throws IOException {
```

Comment:
It looks like this validates records read by the reader against records written by the reader and subsequently read by the reader. I think it should validate the reader and writer separately. I think it should have to parts:

1. Write using generics, read using Flink, and use assertEquals(schema, genericRow, flinkRow)
2. Write using generics, read with Flink, write with Flink, read with generics and use assertEquals(genericRow, endGenericRow).

That way, we're always comparing results against the generics that were originally generated. I think we already have the `assertEquals` code to do it.

Reply:
You testing method is correct, but we don't have `assertEquals(schema, genericRow, flinkRow)` before Junjie's parquet readers & writers patch get in. So I changed to another way to verify the data:
1.  generate `List<records>` by random generater;
2. convert the records to RowData list; 
3.  writer records from step1 to orc file, and reading them into RowData list, and compare with RowData from step2;
4.  write RowData  from step2 into orc file, and reading them into records, and compare with Records from step1.

Reply:
@openinx , Sorry to block you so long. Now it is merged. You might want to take a look.

Reply:
Since your `TestHelpers` got merged,  then I don't have to write the `AssertHelpers` now,  could reuse your work.  Thanks for the work, I've updated the patch.